### ISSUE TITLE
[0.17.0] Regression: Doesn't run `install` script

### ISSUE THREAD

**Issue by STRML** [yarnpkg/yarn#1852]
After upgrading to 0.17.0, I noticed `install` scripts are not running. 

To test, run:

```bash
yarn add nodent-runtime
ls ./node_modules/nodent-runtime/dist
```

There should be an `index.js` built there, but there is not.

The `package.json` for that package:

```json
  "scripts": {
    "test": "node tests",
    "install": "node build.js"
  },
```

This does not happen in 0.16.1.

**Comment by wyze** [comment]
I just tested this on latest master and it does generate a `./node_modules/nodent-runtime/dist/index.js` file.

**Comment by wyze** [comment]
I tested this on 0.17.0 and the scripts run fine for me. What node version an OS are you on?

v6.7.0 on macOS Sierra for me.

**Comment by STRML** [comment]
OSX El Capitan, here's the output:

```
 $ cat package.json
{
  "dependencies": {
    "nodent-runtime": "^3.0.3"
  }
}
~/g/f/footest $ yarn
yarn install v0.17.2
warning No license field
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
✨  Done in 0.17s.
~/g/f/footest $ ll node_modules/nodent-runtime/dist/
total 8
-rw-r--r--  1 user  staff   455B Oct 26 02:38 README.md
~/g/f/footest $ node --version
v6.9.1
```

**Comment by hpurmann** [comment]
macOS Sierra 10.12.1, node 6.9.1

Can't reproduce neither with current `master` (4ce67f3d5da3ba99c3e8718100883e9f145ce125) nor with 0.17.0.

Can you maybe provide a Dockerfile to reproduce?

**Comment by bestander** [comment]
Can't reproduce this on latest either

**Comment by bestander** [comment]
@STRML, can you clear caches?

**Comment by STRML** [comment]
Clearing caches doesn't help - but it appears my `npmrc` is the culprit.

Moving it out fixed the issue. What's perplexing is that when moving back in, the issue remained fixed.

The contents of my global npmrc:

```
_auth=xxxx
init.author.name=My Name
init.author.email=myemail@email.com
init.author.url=http://strml.net/
always-auth=true
spin=false
loglevel=http
@private:registry=http://xx.xx.xx.xx:4873
//xx.xx.xx.xx:4873/:_password=xxxxxx
//xx.xx.xx.xx:4873/:username=myusername
//xx.xx.xx.xx:4873/:email=myemail@email.com
//registry.npmjs.org/:_password=xxxxxx
//registry.npmjs.org/:username=myusername
//registry.npmjs.org/:email=myemail@email.com
//xx.xx.xx.xx:4873/:always-auth=true
//registry.npmjs.org/:always-auth=true
registry=https://registry.npmjs.org/
email=myemail@email.com
progress=false
```

**Comment by STRML** [comment]
I'm not able to reproduce this, even locally after moving my `.npmrc` in and out. Something pretty bizarre there, like a run of yarn without an `.npmrc` present reset something internal which fixed the bug.

If anyone else is experiencing this, working off https://github.com/STRML/yarn-install-lifecycle-bug might help you get to the bottom of it.

**Comment by markstos** [comment]
I recommend closing this issue since the issue can't be reproduced. If someone can reproduce the issue, it can be re-opened.

**Comment by kevinzwhuang** [comment]
@bestander This seems to have been fixed as of v0.17.4 - Also tested for the similar `node-sass` case in #1832: https://github.com/yarnpkg/yarn/issues/1832#issuecomment-261377646
